### ISSUE TITLE
After upgrading to NetBeans9+OpenJDK10 cannot compile

### ISSUE THREAD

**Issue by jsfan3** [codenameone/CodenameOne#2518]
I migrated from Netbeans8.2+JDK8 on Windows to a fresh installation of Netbeans9+OpenJDK10 on Linux. I installed your Codename One plugin 4.4.0

While a default barebones project with a native theme can compile, my Codename One projects don't compile (the problem is the same for different projects and it's the same if I use OpenJDK1.8 instead of OpenJDK10).
Please note that I didn't used the Codename One sources, I tried to compile with the default libraries (I used the "Update Project Libs").

I don't know if this information can be useful: my projects use the CSS support and open the JavaFX window to generate 9-piece borders... and the compiling fails after that.

```
ant -f /home/francesco/Projects/AppBackend -Dnb.internal.action.name=rebuild clean jar
No GUI Entries available
init:
deps-clean:
Updating property file: /home/francesco/Projects/AppBackend/build/built-clean.properties
Deleting directory /home/francesco/Projects/AppBackend/build
refresh-libs-impl:
Deleting directory /home/francesco/Projects/AppBackend/lib/impl
Deleting directory /home/francesco/Projects/AppBackend/native/internal_tmp
clean:
Created dir: /home/francesco/Projects/AppBackend/native/internal_tmp
No GUI Entries available
init:
deps-jar:
Created dir: /home/francesco/Projects/AppBackend/build
Updating property file: /home/francesco/Projects/AppBackend/build/built-jar.properties
Created dir: /home/francesco/Projects/AppBackend/build/classes
Input: /home/francesco/Projects/AppBackend/css/theme.css
Output: /home/francesco/Projects/AppBackend/src/theme.res
InteropFactory: cannot load com.sun.javafx.embed.swing.newimpl.InteropFactoryN
Acquiring lock on CSS checksums file /home/francesco/Projects/AppBackend/.cn1_css_checksums...
Lock obtained
Waiting for web browser
OpenGL Warning: vboxCall failed with VBox status code VERR_BUFFER_OVERFLOW

Opening JavaFX Webview to render some CSS styles
Web browser is available
Releasing lock
java.lang.LinkageError: loader constraint violation: loader (instance of com/codename1/impl/javase/JavaFXLoader$1) previously initiated loading for a different type with name "org/w3c/dom/NodeList"
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1009)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:545)
	at java.base/java.net.URLClassLoader.access$100(URLClassLoader.java:83)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:453)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:447)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:446)
	at com.codename1.impl.javase.JavaFXLoader$1.loadClass(JavaFXLoader.java:314)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
	at com.codename1.designer.css.CSSTheme.getDpi(CSSTheme.java:1716)
	at com.codename1.designer.css.CSSTheme.getBackgroundImage(CSSTheme.java:2012)
	at com.codename1.designer.css.CSSTheme.getBackgroundImages(CSSTheme.java:1778)
	at com.codename1.designer.css.CSSTheme.updateResources(CSSTheme.java:1491)
	at com.codename1.designer.css.CN1CSSCLI.compile(CN1CSSCLI.java:364)
	at com.codename1.designer.css.CN1CSSCLI.main(CN1CSSCLI.java:222)
	at com.codename1.designer.ResourceEditorApp.main(ResourceEditorApp.java:528)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.codename1.impl.javase.JavaFXLoader.runWithJavaFX(JavaFXLoader.java:263)
	at com.codename1.impl.javase.JavaFXLoader.main(JavaFXLoader.java:323)
	at com.codename1.designer.ResourceEditorApp.main(ResourceEditorApp.java:259)
/home/francesco/Projects/AppBackend/build.xml:610: Java returned: 1
BUILD FAILED (total time: 6 seconds)
```

My `build.xml`:
```
<?xml version="1.0" encoding="UTF-8"?>
    <!--build_version=5.1-->
    <!-- 
    This build script was generated by Codename One to build native mobile applications using Java.
    To learn more about Codename One go to https://www.codenameone.com/
    -->
        
    <project name="AppBackend" default="default" basedir=".">
    <description>Builds, tests, and runs the project AppBackend.</description>
    <import file="nbproject/build-impl.xml"/>
    <property file="codenameone_settings.properties"/>
    <property name="automated" value="false" />
    <taskdef name="codeNameOne" classname="com.codename1.build.client.CodeNameOneBuildTask" classpath="CodeNameOneBuildClient.jar"/>
    <taskdef name="prepareTests" classname="com.codename1.build.client.PrepareTestsTask" classpath="CodeNameOneBuildClient.jar"/>
    <taskdef name="installLibs" classname="com.codename1.build.client.InstallLibsTask" classpath="CodeNameOneBuildClient.jar"/>
    <taskdef name="verifyDir" classname="com.codename1.build.client.VerifyDir" classpath="CodeNameOneBuildClient.jar"/>
    <taskdef name="generateGuiSources" classname="com.codename1.build.client.GenerateGuiSources" classpath="CodeNameOneBuildClient.jar"/>
    <taskdef name="initOfflineBuilder" classname="com.codename1.build.client.InitOfflineBuilder" classpath="CodeNameOneBuildClient.jar"/>

    <target depends="init,compile,jar" name="compile-test">
        <mkdir dir="${build.test.classes.dir}" />
        <mkdir dir="${test.src.dir}" />
        <javac destdir="${build.test.classes.dir}"
            encoding="${source.encoding}"
            source="1.8"
            target="1.8"
            bootclasspath="CLDC11.jar" excludes="${excludes}"
            classpath="${javac.classpath}:${build.classes.dir}">
            <src path="${test.src.dir}"/>
        </javac>        
        <prepareTests classesDir="${build.classes.dir}" testClassesDir="${build.test.classes.dir}" seJar="JavaSE.jar" metaDataFile="${build.test.classes.dir}/tests.dat" />
        <jar jarfile="dist/unitTests.jar">
            <fileset dir="${build.classes.dir}"/>
            <fileset dir="${build.test.classes.dir}"/>
            <fileset dir="lib/impl/cls"/>
        </jar>
    </target>

    <target depends="compile-test" name="-do-test-run">        
        <echo>Updating unitTests.jar with native files</echo>
        <jar destfile="dist/unitTests.jar" update="true">
            <fileset dir="native/internal_tmp"/>
        </jar>
        
        <!-- JavaSE.jar is added here since the test execution might fail otherwise for a pre-existing project -->
        <java classpath="dist/unitTests.jar:${javac.test.classpath}:${file.reference.JavaSE.jar}" classname="com.codename1.impl.javase.TestRunner" fork="true">
            <arg value="${codename1.packageName}.${codename1.mainName}" />
        </java>
    </target>
    
    <target depends="-do-test-run" description="Run unit tests." name="test"/>

    <target name="-pre-compile" depends="-cn1-compile-css">
        <echo>Compile is forcing compliance to the supported API's/features for maximum device compatibility. This allows smaller
        code size and wider device support</echo>
        <mkdir dir="build/tmp"/>
        <javac destdir="build/tmp"
            encoding="${source.encoding}"
            source="1.8"
            target="1.8"
            bootclasspath="lib/CLDC11.jar" excludes="${excludes}"
            classpath="${javac.classpath}:${build.classes.dir}">
            <src path="${src.dir}"/>
        </javac>        
    </target>

    <target name="-post-compile">
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/package-info.class"/>
        </delete>
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/package-info.class"/>
        </delete>
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/Thumbs.db"/>
        </delete>
    </target>

    <target name="build-for-ios-device" depends="clean,copy-ios-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            
            targetType="iphone"
            certificate="${codename1.ios.debug.certificate}"
            certPassword="${codename1.ios.debug.certificatePassword}"
            provisioningProfile="${codename1.ios.debug.provision}"
            appid="${codename1.ios.appid}"
            automated="${automated}"
            />
    </target>

    <target name="offline-builder-ios-debug" depends="clean,copy-ios-override,copy-libs,jar,clean-override">
        <initOfflineBuilder />
        <java jar="${selectedOfflineBuilder}">
            <arg value="ios" />
            <arg file="codenameone_settings.properties" />
            <arg file="${dist.jar}" />
        </java>
    </target>

    <target name="offline-builder-ios-release" depends="clean,copy-ios-override,copy-libs,jar,clean-override">
        <initOfflineBuilder />
        <java jar="${selectedOfflineBuilder}">
            <arg value="ios-release" />
            <arg file="codenameone_settings.properties" />
            <arg file="${dist.jar}" />
        </java>
    </target>

    <target name="offline-builder-android" depends="clean,copy-android-override,copy-libs,jar,clean-override">
        <initOfflineBuilder />
        <java jar="${selectedOfflineBuilder}">
            <arg value="and" />
            <arg file="codenameone_settings.properties" />
            <arg file="${dist.jar}" />
            <arg file="gradlePath" />
        </java>
    </target>

    <target name="test-for-ios-device" depends="clean,copy-ios-override,copy-libs,compile-test,clean-override">
        <codeNameOne 
            jarFile="dist/unitTests.jar"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            buildArgs="build.unitTest=1"
            targetType="iphone"
            certificate="${codename1.ios.debug.certificate}"
            certPassword="${codename1.ios.debug.certificatePassword}"
            provisioningProfile="${codename1.ios.debug.provision}"
            appid="${codename1.ios.appid}"
            automated="${automated}"
            />
    </target>

    <target name="test-for-android-device" depends="clean,copy-android-override,copy-libs,compile-test,clean-override">
        <codeNameOne 
            jarFile="dist/unitTests.jar"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            buildArgs="build.unitTest=1"
            targetType="android"
            keystoreAlias="${codename1.android.keystoreAlias}"
            keystore="${codename1.android.keystore}"
            certPassword="${codename1.android.keystorePassword}"
            automated="${automated}"
            />
    </target>

    <target name="test-for-rim-device" depends="clean,copy-rim-override,copy-libs,compile-test,clean-override">
        <codeNameOne 
            jarFile="dist/unitTests.jar"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            buildArgs="build.unitTest=1"
            targetType="rim"
            sigtoolDb="${codename1.rim.signtoolDb}"
            certPassword="${codename1.rim.certificatePassword}"
            sigtoolCsk="${codename1.rim.signtoolCsk}"
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>

     <target name="test-for-j2me-device" depends="clean,j2me-native-theme-init,copy-j2me-native,copy-j2me-override,copy-libs,compile-test,clean-j2me-native,clean-override">
        <codeNameOne
            jarFile="dist/unitTests.jar"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            targetType="me"            
            buildArgs="build.unitTest=1"
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>

    
    <target name="build-for-ios-device-release" depends="clean,copy-ios-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            appStoreBuild="true"
            production="true"
            targetType="iphone"
            certificate="${codename1.ios.release.certificate}"
            certPassword="${codename1.ios.release.certificatePassword}"
            provisioningProfile="${codename1.ios.release.provision}"
            appid="${codename1.ios.appid}"
            automated="${automated}"
            />
    </target>

    <target name="copy-libs" depends="init">
        <copy todir="${build.classes.dir}">
            <fileset dir="lib/impl/cls" />
        </copy>
    </target>

    <target name="copy-android-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_android.ovr"/>
            <fileset dir="override" includes="**/*_android-phone.ovr"/>
            <fileset dir="override" includes="**/*_android-tab.ovr"/>
            <fileset dir="override" includes="**/*_phone.ovr"/>
            <fileset dir="override" includes="**/*_tablet.ovr"/>
        </copy>
        <copy todir="${build.classes.dir}">
            <fileset dir="native/android" includes="**/*"/>
            <fileset dir="lib/impl/native/android" includes="**/*"/>
        </copy>
    </target>
    
    <target name="copy-ios-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_ios.ovr"/>
            <fileset dir="override" includes="**/*_iphone.ovr"/>
            <fileset dir="override" includes="**/*_ipad.ovr"/>
            <fileset dir="override" includes="**/*_phone.ovr"/>
            <fileset dir="override" includes="**/*_tablet.ovr"/>
        </copy>
        <copy todir="${build.classes.dir}">
            <fileset dir="native/ios" includes="**/*"/>
            <fileset dir="lib/impl/native/ios" includes="**/*"/>
        </copy>
    </target>
        
    <target name="copy-javascript-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <mkdir dir="native/javascript" />
        <mkdir dir="lib/impl/native/javascript" />
        <copy todir="${build.classes.dir}">
            <fileset dir="native/javascript" includes="**/*"/>
            <fileset dir="lib/impl/native/javascript" includes="**/*"/>
        </copy>
    </target>
    
    <target name="copy-rim-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_rim.ovr"/>
            <fileset dir="override" includes="**/*_phone.ovr"/>
        </copy>
        <copy todir="${build.classes.dir}">
            <fileset dir="native/rim" includes="**/*"/>
            <fileset dir="lib/impl/native/rim" includes="**/*"/>
        </copy>
    </target>

    <target name="copy-desktop-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <mkdir dir="lib/impl/native/javase" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_tablet.ovr"/>
            <fileset dir="override" includes="**/*_desktop.ovr"/>
        </copy>
        <copy todir="${build.classes.dir}">
            <fileset dir="native/javase" includes="**/*"/>
            <fileset dir="lib/impl/native/javase" includes="**/*"/>
        </copy>
    </target>

    <target name="copy-j2me-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_me.ovr"/>
            <fileset dir="override" includes="**/*_phone.ovr"/>
        </copy>
        <copy todir="${build.classes.dir}">
            <fileset dir="native/j2me" includes="**/*"/>
            <fileset dir="lib/impl/native/j2me" includes="**/*"/>
        </copy>
    </target>

    <target name="clean-override">
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.ovr"/>
            <fileset dir="${build.classes.dir}" includes="**/*.java"/>
            <fileset dir="${build.classes.dir}" includes="**/*.m"/>
            <fileset dir="${build.classes.dir}" includes="**/*.h"/>
        </delete>
    </target>
    
    <target name="build-for-android-device" depends="clean,copy-android-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            
            targetType="android"
            keystoreAlias="${codename1.android.keystoreAlias}"
            keystore="${codename1.android.keystore}"
            certPassword="${codename1.android.keystorePassword}"
            automated="${automated}"
            />
    </target>

    <target name="build-for-javascript" depends="clean,copy-javascript-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            
            targetType="javascript"
            automated="${automated}"
            />
    </target>    
    
    <target name="war" depends="clean,copy-javascript-override,copy-libs,jar,clean-override" description="Create .war file with Javascript build">
        <antcall target="build-for-javascript">
            <param name="automated" value="true"/>
        </antcall>
        <basename property="dist.base" file="${dist.jar}" suffix=".jar"/>
        <property name="dist.war" value="dist/${dist.base}.war"/>
        <mkdir dir="build/result" />
        <unzip src="dist/result.zip" dest="build/result" />
        <first id="first">
            <fileset dir="build/result" includes="*.war"/>
        </first>
        <copy file="${toString:first}" tofile="${dist.war}" overwrite="true"/>
        <delete dir="build/result"/>
        <delete file="dist/result.zip"/>
    </target>
    <target name="check-webapp-runner">
        <echo>Checking Webapp Runner...</echo>
        <available file="${user.home}/.codenameone/webapp-runner.jar" property="webapprunner.present" value="true"/>
        <echo message="Webapp Runner present? : ${webapprunner.present}"/>
    </target>
    <target name="init-webapp-runner" depends="check-webapp-runner" unless="webapprunner.present">
        <mkdir dir="${user.home}/.codenameone"/>
        <get src="https://github.com/codenameone/cn1-binaries/raw/master/javascript/webapp-runner.jar" dest="${user.home}/.codenameone/webapp-runner.jar"/>
    </target>
    <target name="check-war" depends="init">
        <echo>Checking War</echo>
        <basename property="dist.base" file="${dist.jar}" suffix=".jar"/>
        <property name="dist.war" value="dist/${dist.base}.war"/>
        <available file="${dist.war}" property="war.present" value="true"/>
        <echo message="War file present? ${war.present}"/>
    </target>
    <target name="init-war" depends="check-war" unless="war.present">
        <antcall target="war"/>
    </target>
    <target name="run-war" depends="init-war,init-webapp-runner" description="Run as Web App on Localhost using embedded Tomcat">
        <property name="port" value="0"/>
        <java jar="${user.home}/.codenameone/webapp-runner.jar" fork="true" failonerror="true">
            <arg value="${dist.war}"/>
            <arg value="--port"/>
            <arg value="${port}"/>
            <arg value="--open-in-browser"/>
            <arg value="--temp-directory"/>
            <arg value="dist/webapp"/>
        </java>
    </target>
    <target name="build-for-rim-device" depends="clean,copy-rim-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"

            targetType="rim"
            sigtoolDb="${codename1.rim.signtoolDb}"
            certPassword="${codename1.rim.certificatePassword}"
            sigtoolCsk="${codename1.rim.signtoolCsk}"
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>

    <target name="build-for-mac-os-x-desktop" depends="clean,copy-desktop-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            certificate="${codename1.desktop.mac.certificate}"
            certPassword="${codename1.desktop.mac.certificatePassword}"
            targetType="desktop_macosx"
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>
    
    <target name="build-for-windows-desktop" depends="clean,copy-desktop-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"

            targetType="desktop_windows"
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>

    <target name="build-for-j2me-device" depends="clean,j2me-native-theme-init,copy-j2me-native,copy-j2me-override,copy-libs,jar,clean-j2me-native,clean-override">
        <codeNameOne
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            targetType="me"            
            automated="${automated}"
            />
        <delete>
            <fileset dir="${build.classes.dir}" includes="**/*.class"/>
        </delete>
    </target>
      
    <target name="j2me-native-theme-init">
        <available file="${codename1.j2me.nativeTheme}" property="j2me-native-theme.available" />
    </target>
    
    <target name="copy-j2me-native" if="j2me-native-theme.available">
        <copy file="${codename1.j2me.nativeTheme}" tofile="src/nativeJ2METheme.res" />
    </target>

    <target name="clean-j2me-native" if="j2me-native-theme.available">
        <delete file="src/nativeJ2METheme.res" />
    </target>
      
    <target name="-pre-init">
        <property name="application.args" value="${codename1.packageName}.${codename1.mainName}"/>
        <property name="javac.source" value="1.8" />
        <property name="javac.target" value="1.8" />
        <mkdir dir="native/javase" />
        <mkdir dir="native/android" />
        <mkdir dir="native/internal_tmp" />
        <mkdir dir="native/ios" />
        <mkdir dir="native/j2me" />
        <mkdir dir="native/rim" />
        <mkdir dir="native/win" />
        <mkdir dir="native/javascript" />
        <mkdir dir="lib/impl/cls" />
        <mkdir dir="lib/impl/stubs" />
        <mkdir dir="lib/impl/native" />
        <mkdir dir="lib/impl/native/android" />
        <mkdir dir="lib/impl/native/ios" />
        <mkdir dir="lib/impl/native/j2me" />
        <mkdir dir="lib/impl/native/rim" />
        <mkdir dir="lib/impl/native/win" />
        <mkdir dir="lib/impl/native/javascript" />
        <installLibs lib="lib" dest="lib/impl" />
        <generateGuiSources srcDir="src" encoding="UTF-8" guiDir="res/guibuilder"  />
    </target>

    <target name="-post-jar">
        <mkdir dir="native/javase" />
        <mkdir dir="native/internal_tmp" />
        <mkdir dir="lib/impl/native/javase" />
        <javac destdir="native/internal_tmp"
            encoding="${source.encoding}"
            source="1.8"
            target="1.8"
            classpath="${run.classpath}:${build.classes.dir}">
            <src path="native/javase"/>
            <src path="lib/impl/native/javase"/>
        </javac>
        <copy todir="native/internal_tmp">
            <fileset dir="native/javase" excludes="*.java,*.jar"/>
            <fileset dir="lib/impl/native/javase" excludes="*.java,*.jar"/>
        </copy>        
    </target>

    <target name="-post-clean" depends="refresh-libs-impl">
        <delete dir="native/internal_tmp"/>
    </target>

    <target name="refresh-libs" depends="refresh-libs-impl,jar">
    </target>

    <target name="refresh-libs-impl">
        <delete dir="lib/impl" />
        <installLibs lib="lib" dest="lib/impl" />        
    </target>

    <target name="copy-windows-override">
        <mkdir dir="${build.classes.dir}" />
        <mkdir dir="override" />
        <copy todir="${build.classes.dir}">
            <fileset dir="override" includes="**/*_win.ovr"/>
            <fileset dir="override" includes="**/*_win-phone.ovr"/>
            <fileset dir="override" includes="**/*_win-tab.ovr"/>
            <fileset dir="override" includes="**/*_phone.ovr"/>
            <fileset dir="override" includes="**/*_tablet.ovr"/>
        </copy>
        <mkdir dir="native/win" />
        <copy todir="${build.classes.dir}">
            <fileset dir="native/win" includes="**/*"/>
            <fileset dir="lib/impl/native/win" includes="**/*"/>
        </copy>
    </target>

    <target name="build-for-win-device" depends="clean,copy-windows-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            
            targetType="win"
            automated="${automated}"
            />
    </target>
                
    <target name="build-for-windows-device" depends="clean,copy-windows-override,copy-libs,jar,clean-override">
        <codeNameOne 
            jarFile="${dist.jar}"
            displayName="${codename1.displayName}"
            packageName = "${codename1.packageName}"
            mainClassName = "${codename1.mainName}"
            version="${codename1.version}"
            icon="${codename1.icon}"
            vendor="${codename1.vendor}"
            subtitle="${codename1.secondaryTitle}"
            
            certificate="${codename1.windows.certificate}"
            certPassword="${codename1.windows.certificatePassword}"

            targetType="windows"
            automated="${automated}"
            />
    </target>
                
    <target name="webservice-wizard">
        <java classpath="CodeNameOneBuildClient.jar" classname="com.codename1.ws.wizard.WSWizard" dir="." fork="true" />
    </target>
    
    <target name="-cn1-compile-css" if="codename1.cssTheme">
        <java jar="${user.home}/.codenameone/designer_1.jar" failonerror="true">
            <jvmarg value="-Dcli=true"/>
            <arg value="-css"/>
            <arg file="css/theme.css"/>
            <arg file="src/theme.res"/>
        </java>
    </target>
</project>
```

**Comment by jsfan3** [comment]
SOLVED.

**Short story**: the solution is to replace OpenJDK10 with OracleJDK10

**Long story**: it's not only a problem of JDK. There are several issues...

In the reported log, it's written: «vboxCall failed with VBox status code VERR_BUFFER_VERFLOW». It's a bug of VirtualBox (https://forums.virtualbox.org/viewtopic.php?f=2&t=85559), as workaround I disabled the 3D acceleration.

The second issue was about: «java.lang.LinkageError: loader constraint violation: loader (instance of com/codename1/impl/javase/JavaFXLoader$1) previously initiated loading for a different type with name "org/w3c/dom/NodeList"»

To solve the second issue, I created new Netbeans projects (in a different location from the other projects) with the same project names, same project packages and same main classes. After that, I copied from the old projects to the new projects these dirs and files:

```
<projectDir>/css
<projectDir>/src
<projectDir>/icon.png
<projectDir>/lib/<CN1Extensions> (.jar and .ver files)
```

Now the projects compile.

Third issue: they don't run because the same "java.lang.LinkageError". It seems the same previous issue but the previous occurs at compile time, this occurs at runtime.

Fourth issue: I removed the build and dist directories, I changed in projects properties the Java Platform from JDK10 to JDK1.8
(they are OpenJDK versions even if they are reported as JDK), but when I tried to run them I got an infinite log of CSS errors.

I installed Netbeans 8.2 (with OpenJDK 10), I opened the same projects... but Netbeans 8.2 is not compatible with OpenJDK 10, it doesn't work.
Note that Netbeans 8.2 and Netbeans 9.0 store their preferences and plugins in different folders, so there is no conflit.

Finally, I replaced OpenJDK10 with OracleJDK10 following these instructions:
https://stackoverflow.com/questions/49507160/how-to-install-jdk-10-under-ubuntu
(Please note that these instruction doesn't leave both OpenJDK and OracleJDK, but they remove OpenJDK automatically during the installation of OracleJDK)

I set OracleJDK10 as the Java Platform of the projects. I compiled them. I pressed run: after few attemps, now they work!!! (But, fifth issue: I needed to delete the files theme.css.checksums, I replaced the theme.res files with empty new theme.res files taken from a new project with native theme and I run the projects more than two times to get them working).

So, in conclusion: the CSS support on OpenJDK10 seems broken, while it seems correct on OracleJDK10.

I close this issue because there are too many issues inside. Maybe you can work on a better working of the CSS support.
